components/vocabularies/oncotree/api/src/main/java/org/phenotips/vocabulary/internal/solr/OncoTree.java

```diff
@@ -0,0 +1,322 @@
+    }
+
+    /**
+     * Processes a CSV {@code row}, and adds the relevant fields to a collection of {@code data}.
+     *
+     * @param row a {@link CSVRecord} containing data for one path (from root to leaf) of the OncoTree
+     * @param data a collection of {@link SolrInputDocument} objects where data from {@code row} will be written
+     */
+    private void processDataRow(@Nonnull final CSVRecord row, @Nonnull final Collection<SolrInputDocument> data)
+    {
+        final SolrInputDocument doc = new SolrInputDocument();
+
+        for (int i = 0; i < row.size(); i++) {
+            final String label = COLUMNS[i];
+            final String value = row.get(i);
+            applyWriteStrategies(doc, label, value);
+        }
+        data.add(doc);
+    }
+
+    /**
+     * Writes the {@code label} and {@code value} data into the {@code doc}, as per some predefined write strategies.
+     *
+     * @param doc the {@link SolrInputDocument} where the data will be written
+     * @param label the property label
+     * @param value the property value
+     */
+    private void applyWriteStrategies(
+        @Nonnull final SolrInputDocument doc,
+        @Nonnull final String label,
+        @Nullable final String value)
+    {
+        final Collection<OncoTreePropertyWriteStrategy> strategies = STRATEGIES.get(label);
+        for (final OncoTreePropertyWriteStrategy strategy : strategies) {
+            strategy.execute(doc, label, value);
+        }
+    }
+
+    /**
+     * Sets up the CSV parser with tab-delimited format, and provided {@link #COLUMNS column names}.
+     *
+     * @return a {@link CSVRecord parser}
+     */
+    private CSVFormat setupCSVParser()
+    {
+        return CSVFormat.TDF.withHeader(COLUMNS).withSkipHeaderRecord();
+    }
+
+    /**
+     * Returns the prefix for the vocabulary terms belonging to the OncoTree vocabulary.
+     *
+     * @return a prefix for the OncoTree vocabulary terms
+     */
+    private String getTermPrefix()
+    {
+        return "ONCO";
+    }
+
+    @Override
+    protected String getCoreName()
+    {
+        return getIdentifier();
+    }
+
+    @Override
+    public String getIdentifier()
+    {
+        return "onco";
+    }
+
+    @Override
+    public String getName()
+    {
+        return "OncoTree";
+    }
+
+    @Override
+    public Set<String> getAliases()
+    {
+        final Set<String> aliases = new HashSet<>();
+        aliases.add(getName());
+        aliases.add(getIdentifier());
+        aliases.add(getTermPrefix());
+        return Collections.unmodifiableSet(aliases);
+    }
+
+    @Override
+    public String getDefaultSourceLocation()
+    {
+        return SOURCE;
+    }
+
+    @Override
+    public String getWebsite()
+    {
+        return "http://oncotree.mskcc.org/oncotree/";
+    }
+
+    @Override
+    public String getCitation()
+    {
+        // FIXME: Citation needs to be added. How do they want to be credited?
+        return "";
```

Comment:
Note: This will need to be changed once licensing is confirmed. I'm not sure how they would like to be credited.